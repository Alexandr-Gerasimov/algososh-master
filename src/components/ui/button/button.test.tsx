import renderer from "react-test-renderer";
import { Button } from "./button";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Buttons", () => {
  it("Кнопки с текстом", () => {
    const tree = renderer.create(<Button text="Развернуть" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопки без текста", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Заблокированные кнопки", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопки с индикацией загрузки", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Button is working", () => {
  it("Вызов колбека при клике на кнопку", () => {
    window.alert = jest.fn();
    render(<Button text="Click" onClick={() => alert("Работает")} />);
    const button = screen.getByText("Click");
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith("Работает");
  });
});
