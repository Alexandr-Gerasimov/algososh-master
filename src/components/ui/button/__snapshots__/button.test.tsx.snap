// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Заблокированные кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Buttons Кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Buttons Кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Buttons Кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;
